### ISSUE TITLE
Bulk saving products using WC_Product::save() takes too long 

### ISSUE THREAD

**Issue by pirex360** [woocommerce/woocommerce#27249]
I'm writing a plugin to WP->Woocomerce to import products from a custom Webservice.

But saving products using WC_Product::save() method takes about 15/20 seconds per product in my case.
Note that i also save the images... about 4 images (40Kb to 60Kb each) per product and wordpress is also slow to pull an external image to his library.

Without images, only text and simple product : my overall average is 3-4 seconds per product.

I need to optimize this saving to WP database, I read the external webservice, its about 3MB Json file in 4-5 seconds... but saving the products...is a nightmare, i dont get it !!! I have 1000 products only.

**To import 1000 products is around 266 minutes** 

Some workarounds to gain some time reduction:

- when creating product categories use first : `wp_defer_term_counting(true);` - this disable interactive counting for products in product categories (save me 1a2 seconds for each product)
- only after adding all the products, count and update the product categories numbers.
- when import a image... use :`add_filter('jpeg_quality', function($arg){return 100;});`
- this stops image manipulation by WP, but you will not get lower image size optimization
- after all products import i put back in default : `add_filter('jpeg_quality', function($arg){return 80;});`
- I created a javascript pool to import 2 products at the time in async way, if i increase the number of products to 4 ou 5 : mariadb server overloads and get timeout

What i'm doing wrong ? Or is anything wrong with your method of saving products from your class.

**Comment by juliaamosova** [comment]
Hi @pirex360,

While our goal is to address all the issues reported in this repository, GitHub should be treated as a place to report confirmed bugs only.

The type of issue you submitted looks like a custom code related question. Due to the number of bug reports being submitted, we are prioritizing work on them and are not able to attend to custom code questions. 

The best place to get help further is in the [WooCommerce Community Slack](https://woocommerce.com/community-slack/) in the `#developers` channel. 

If you confirm the bug, please open a new issue and provide us with clear steps to reproduce it.

I am closing this issue for now. If I missed something, please leave a comment and we will take a second look.

**Comment by pirex360** [comment]
Ok, thanks for the feedback. 
I tought that the extremly slow save time, was a bug or some lack of optimization.

Thanks again for your recomendation.
